zerver/views/upload.py

```diff
@@ -71,5 +71,10 @@ def upload_file_backend(request, user_profile):
     if ((settings.MAX_FILE_UPLOAD_SIZE * 1024 * 1024) < user_file._get_size()):
         return json_error(_("File Upload is larger than allowed limit"))
 
+    if not isinstance(user_file.name, str):
+        # This check makes sure that the type of the file name is not
+        # Python 2 Unicode.
+        user_file.name = user_file.name.encode('ascii')
```

Comment:
Can you expand this comment a bit more to explain why the new logic is correct?  I don't totally follow and I think it'll be especially unclear to future developers who didn't see the original issue.  In particular, it's not clear to me why the logic will be correct in Python 3 where `str` is `unicode`.